### DISCUSSION TITLE
Support for Box-Jenkins model structure

### DISCUSSION THREAD

**Discussion by Hofsmo** [baggepinnen/ControlSystemIdentification.jl#153] [Ideas]
Hi,
I was wondering if there are any plans on adding support for the Box-Jenkins model structure? For some of my previous research I used the implemenentation from the System Identification Toolbox in MATLAB quite a lot, but I never found a good open source alternative.

**Comment by baggepinnen** [comment]
Hi!

Not really, a black-box statespace model structure can represent any Box-Jenkins structured model, so I haven't seen any good-enough motivation to adding additional support for those Box-Jenkins and instead recommend using `newpem` to estimate a general statespace model. With a statespace model from `newpem`, you can decompose this model into the part driven by input and the part driven by noise using `innovation_form`, the first example here demonstrates that [Validation](https://baggepinnen.github.io/ControlSystemIdentification.jl/dev/validation/).


Having said that, the [`structured_pem`](https://baggepinnen.github.io/ControlSystemIdentification.jl/dev/ss/#ControlSystemIdentification.structured_pem) method is able to estimate any structured linear model, but to use it you'd have to take the parameter vector and turn that into your BJ model and finally convert that into statespace form using `ss` in the user-provided [`constructor` function](https://baggepinnen.github.io/ControlSystemIdentification.jl/dev/ss/#ControlSystemIdentification.structured_pem).

**Reply by Hofsmo** [reply]
Many thanks for the quick reply. I don't have any experience with state space identification, but I will check it out :) .

**Reply by baggepinnen** [reply]
Please be encouraged to reach out if any questions appear!
